### PR TITLE
Add ci and release github actions for easier maintenance

### PR DESCRIPTION
**Pull request by mbachry**
Linux and Mac binary wheels were handled by travis builds, but it seems that travis is currently broken. Instead of fixing it, I suggest a single github action that builds wheels for all three platforms (thus replacing both travis and appveyor). The action is triggered by pushing a git tag with `v` prefix (eg. `v2.1.0`) - this should be the only operation required to make a full release.

Some changes:
* `build.yml` action that builds and uploads a release on a new git tag; regular pushes upload to test.pypi.org (matches current behavior)
* added [setuptools_scm](https://github.com/pypa/setuptools_scm/), so you don't have to edit `version.py` (the version is picked automatically from the current git tag)
* a simple `ci.yml` for running tests

Here's my test release: https://test.pypi.org/project/hiredis-mytest/0.0.0.dev0/#files (all currently supported platforms are there)

Note that `TEST_PYPI_API_TOKEN` and `PYPI_API_TOKEN` [pypi secrets](https://pypi.org/help/#apitoken) would have to be created in github repo settings for pypi uploads to work.

### CODE REVIEW

**Comment by jlvrhee** [comment]
Is there anyone who can merge this pull request, so we get an updated wheel in pypi?

Thanks in advance

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        os: [ubuntu-20.04, windows-2019, macos-10.15]
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Set test tag
+        if: ${{ !startsWith(github.event.ref, 'refs/tags/v') }}
+        run: git tag -f v0.0.0-dev0
```
Comment (rth) [inline .github/workflows/build.yml:20]: We shouldn't forget to remove it before merging.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Set test tag
+        if: ${{ !startsWith(github.event.ref, 'refs/tags/v') }}
+        run: git tag -f v0.0.0-dev0
```
Comment (rth) [inline .github/workflows/build.yml:42]: Same comment.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+      - name: Build sdist
+        run: pipx run build --sdist
```
Comment (rth) [inline .github/workflows/build.yml:45]: Let's maybe avoid pipx,
```
pip install build
python -m build --sdist
```
is the more standard way.

**Review by rth — COMMENTED** [review]
This looks really good. Thanks @mbachry . Now we need to find a maintainer to review/merge it.

Also Github Actions didn't trigger. Either it needs to be approved by a maintainer or I found that manually adding some starting setup in the "Actions" tab is usually sufficient to make it enabled.

**Comment by rth** [comment]
Yes, makes sense. Looks like it's blocked by https://github.com/redis/hiredis-py/issues/63 for now.

**Pull request closed without merging**